tests/eas/load_tracking.py

```diff
@@ -537,6 +540,206 @@ def test_load_avg_behaviour(self, experiment, tasks):
+        :param tasks: json description of the tasks
+        :type tasks: dict
+        :return a list of CPUs on which tasks are running
+        """
+        cpus = []
+        phases = tasks[tasks.keys()[0]]['phases'].keys()
+        for phase in phases:
+            for task in tasks.keys():
+                [cpu] = tasks[task]['phases'][phase]['cpus']
+                if cpu not in cpus:
+                    cpus.append(cpu)
+
+        return cpus
+
+    def _get_stable_range(self, df, tasks, phase_id):
+        """
+        Return a start and end interval for which the utilization signal should
+        be stabilized.
+        :param df: data frame of the sched_switch events
+        :type df: DataFrame
+
+        :param tasks: json description of the tasks
+        :type tasks: dict
+
+        :param phase_id: index (starting from 0) of the phase for which the
+                         stable range is computed
+        :type phase: int
+        :return a start time and a end time where the utilization signal should
+                be stabilized.
+        """
+        task_names = tasks.keys()
+        phase_duration_s = 0
+        phase_names = tasks[task_names[0]]['phases'].keys()
+        df = df[df.next_comm.isin(task_names)]
+        start = df.index[1]
+        # Compute the total duration of the previous phases to get the start
+        # time of the given phase
+        for phase in (phase_names[: phase_id+1]):
+            start += phase_duration_s
+            loop = tasks[task_names[0]]['phases'][phase]['loop']
+            period_us = tasks[task_names[0]]['phases'][phase]['timer']['period']
+            phase_duration_s = (loop * period_us) / 1e6
+
+        end = start + phase_duration_s - UTIL_AVG_CONVERGENCE_TIME
+        start += UTIL_AVG_CONVERGENCE_TIME
+
+        return start, end
+
+    def _test_util_per_cpu(self, experiment, tasks):
+        trace = self.get_trace(experiment)
+        if 'sched_switch' not in trace.available_events:
+            raise ValueError('No sched_switch events. '
+                             'Does the kernel support them?')
+        if 'sched_load_cfs_rq' not in trace.available_events:
+            raise ValueError('No sched_load_cfs_rq events. '
+                             'Does the kernel support them?')
+
+        # Load the JSON tasks description
+        tasks_desc = json.load(open(experiment.wload.params['custom']),
+                               object_pairs_hook=OrderedDict)['tasks']
+        sw_df = trace.data_frame.trace_event('sched_switch')
+        util_df = trace.data_frame.trace_event('sched_load_cfs_rq')
+
+        phases = tasks_desc[tasks_desc.keys()[0]]["phases"].keys()
+        cpus = self._get_cpus(tasks_desc)
+
+        msg = 'Saw util {} on cpu {}, expected {} during phase {}'
+        for j in range(len(phases)):
+            # Compute the interval where the signal is stable for a phase
+            stable_start, stable_end = self._get_stable_range(sw_df,
```

Comment:
Would use just `start` and `end`... we know the context or you can make it explicit with just a comment.

Reply:
Done